src/main/java/org/sagebionetworks/bridge/hibernate/HibernateSchedule2Dao.java

```diff
@@ -98,61 +118,125 @@ final void setHibernateHelper(HibernateHelper hibernateHelper) {
     @Override
     public Schedule2 createSchedule(Schedule2 schedule) {
         checkNotNull(schedule);
-        
-        hibernateHelper.create(schedule);
+
+        Timeline timeline = Scheduler.INSTANCE.calculateTimeline(schedule);
+        List<TimelineMetadata> metadata = timeline.getMetadata();
+
+        hibernateHelper.executeWithExceptionHandling(schedule, (session) -> {
+            // batch these operations. Improves network performance
+            session.setJdbcBatchSize(batchSize);
+
+            session.save(schedule);
+
+            if (!metadata.isEmpty()) {
+                Stopwatch stopwatch = Stopwatch.createStarted();
+                // Batch deleting/recreating rather than updating is 2-3 orders of magnitude faster.
+                NativeQuery<?> query = session.createNativeQuery(DELETE_TIMELINE_RECORDS);
+                query.setParameter(SCHEDULE_GUID, schedule.getGuid());
+                query.executeUpdate();
```

Comment:
You need to delete these records every time!

Reply:
Removed the test for an empty collection, verified with manual testing this is not an issue.